homeassistant/components/unifiprotect/event.py

```diff
@@ -69,13 +162,24 @@ def _async_event_entities(
     data: ProtectData,
     ufp_device: ProtectAdoptableDeviceModel | None = None,
 ) -> list[ProtectDeviceEntity]:
+    # Define a mapping between the description keys and corresponding entity classes
+    entity_mapping: dict[str, type[ProtectDeviceEntity]] = {
+        "nfc": ProtectDeviceNFCEventEntity,
+        "fingerprint": ProtectDeviceFingerprintEventEntity,
+        "ring": ProtectDeviceRingEventEntity,
+    }
+
     entities: list[ProtectDeviceEntity] = []
+
     for device in data.get_cameras() if ufp_device is None else [ufp_device]:
-        entities.extend(
-            ProtectDeviceEventEntity(data, device, description)
-            for description in EVENT_DESCRIPTIONS
-            if description.has_required(device)
-        )
+        for description in EVENT_DESCRIPTIONS:
```

Comment:
This could likely be collapsed back into the generator expression inside the extend